src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -62,11 +63,21 @@ class RoomSchemaLocationWorkaround implements Workaround {
             roomSchemaMergeLocations = roomExtension.roomSchemaMergeLocations
         }
 
-        JavaCompileWorkaround javaCompileRoomTask = JavaCompileWorkaround.create(project, roomExtension, mergeTask)
 
-        project.plugins.withId("kotlin-kapt") {
-            KaptWorkaround.create(project, roomExtension, mergeTask)
-            javaCompileRoomTask.javaCompileSchemaGenerationEnabled = false
+        // We favor ksp over kapt and apt in projects with multiple processors configured. If Ksp is applied and using
+        // the room-compiler, kapt and apt workarounds won't be applied.
+        if (project.configurations.find { it.name == "ksp" } != null) {
```

Comment:
@runningcode @bigdaz removed the `afterEvaluate` block and removing the responsibility of checking the ksp/room-compiler from the workaround implementations. With `withDependencies` I can verify if the project is using `room-compiler`/`ksp`